specifyweb/frontend/js_src/lib/components/WorkBench/index.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { useAsyncState } from '../../hooks/useAsyncState';
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { useErrorContext } from '../../hooks/useErrorContext';
+import { commonText } from '../../localization/common';
+import { wbText } from '../../localization/workbench';
+import { ajax } from '../../utils/ajax';
+import { f } from '../../utils/functools';
+import type { GetSet } from '../../utils/types';
+import { className } from '../Atoms/className';
+import { LoadingContext } from '../Core/Contexts';
+import { LoadingScreen } from '../Molecules/Dialog';
+import { useMenuItem } from '../Header/MenuContext';
+import { treeRanksPromise } from '../InitialContext/treeRanks';
+import { Dialog } from '../Molecules/Dialog';
+import { NotFoundView } from '../Router/NotFoundView';
+import type { Dataset } from '../WbPlanView/Wrapped';
+import { WbView } from './WbView';
+
+const fetchTreeRanks = async (): Promise<true> => treeRanksPromise.then(f.true);
+
+const fetchDataSet = async (
+  datasetId: number | undefined
+): Promise<Dataset | undefined> =>
+  typeof datasetId === 'number'
+    ? ajax<Dataset>(`/api/workbench/dataset/${datasetId}/`, {
+        headers: { Accept: 'application/json' },
+      }).then(({ data }) => data)
+    : undefined;
+
+// BUG: intercept 403 (if dataset has been transferred to another user)
+function useDataSet(
+  datasetId: number | undefined
+): GetSet<Dataset | undefined> {
+  return useAsyncState(
+    React.useCallback(async () => fetchDataSet(datasetId), [datasetId]),
+    true
+  );
+}
+
+export function WorkBench(): JSX.Element {
```

Comment:
could you please reorder this file a bit:
generally, the main thing/exported thing/main component (WorkBench in this case) should be first in the file, and then, all the sub functions/sub components/variables that component uses should be defined below this main thing, ideally, in the order in which they are accessed in this main thing.

so you would have WorkBench, then fetchTreeRanks, useDataSet, fetchDataSet

Pros:
- since WorkBench component is the main thing this file is about, it should be first to represent correctly what this file is about
- if you add js doc comments to the main thing (WorkBench component), that comment would be near the very top of the file, thus immediately visible as soon as the fil is opened. Such comment is likely of high relevance for anyone opening this file for the first time
- When reading the code, having sub components/sub functions defined below the main thing in the order they are used makes code easier to read - when you are reading, you don't have to jump and down in the file as much, as the things are defined roughly in the order in which the code would be executing (this also makes the code jump a bit less when going one by one in the debugger)

this is not a hard rule, and you are welcome to order things the way you think is best, but this is what we found is good heuristic most of the time for making code more readable.
one case where this is not possible is if the main thing in your file is a variable/object rather than a function (since variables are not hoisted, unlike functions, if object is using a subobject, that subobject has to be defined above it)